File: Leetcode/PY/Leetcode66-加一.py
from typing import List
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        n,car = len(digits),1
        for i in range(n - 1,-1,-1):
            cur = car + digits[i]
            digits[i] = cur % 10
            car = cur // 10
        if car:
            digits.insert(0,1)
        return digits

File: Leetcode/sbgo/Leetcode66-加一.go
package sbgo

func plusOne(digits []int) []int {
	n, car := len(digits), 1
	for i := n - 1; i >= 0; i-- {
		cur := car + digits[i]
		digits[i] = cur % 10
		car = cur / 10
	}
	if car != 0 {
		digits = append([]int{1}, digits...)
	}
	return digits
}
